inference-engine/src/gna_plugin/frontend/model_quantizer.hpp

```diff
@@ -167,13 +168,16 @@ class ModelQuantizer {
             if (infiniteLoopHistory.empty()) {
                 infiniteLoopCount++;
             } else {
-                if (infiniteLoopCount > 0 &&
-                    (infiniteLoopHistory.size()%infiniteLoopPattern.size() == 0 || sf.allLayersProcessed()) &&
-                    !std::equal(infiniteLoopHistory.begin() + (infiniteLoopHistory.size() - infiniteLoopPattern.size()),
-                        infiniteLoopHistory.end(), infiniteLoopPattern.begin())) {
-                    infiniteLoopCount = 0;
-                    infiniteLoopPattern.clear();
-                    gnalog() << "infinite loop fixed\n";
+                if (infiniteLoopCount > 0 && (infiniteLoopHistory.size() % infiniteLoopPattern.size() == 0 || sf.allLayersProcessed())) {
+                    if (infiniteLoopHistory.size() < infiniteLoopPattern.size() ||
+                        // TODO: analyse why infiniteLoopHistory.size() < infiniteLoopPattern.size() happen in the following test:
+                        // smoke_InsertCopy/InsertCopyBeforeSelfConcatTest.CompareWithRefs/
+                        // IS=(1.16)_IN=2_CN=1_axis=1_netPRC=FP32_inPRC=UNSPECIFIED_outPRC=UNSPECIFIED_inL=ANY_outL=ANY_trgDev=GNA
+                        !std::equal(infiniteLoopPattern.rbegin(), infiniteLoopPattern.rend(), infiniteLoopHistory.rbegin())) {
+                        infiniteLoopCount = 0;
+                        infiniteLoopPattern.clear();
+                        gnalog() << "infinite loop fixed\n";
+                    }
```

Comment:
@dmitriikhurtin I see you were changing this file recently.
I was getting SEGFAULT because in the mentioned test I was getting the state `infiniteLoopHistory.size() < infiniteLoopPattern.size()` so the iterator was invalid, so I had to add this W/A to my PR. 
Could you help to review this part?

Reply:
We also can have a talk if needed.

Reply:
@sirzabek FYI